[Eval] Evals for trivia questions of US states and capital.

fengttt opened the pull request:
Trivia questions of US states and capital with some word puzzle twists.

# Thank you for contributing an eval! ♥️

🚨 Please make sure your PR follows these guidelines, **failure to follow the guidelines below will result in the PR being closed automatically**. Note that even if the criteria are met, that does not guarantee the PR will be merged nor GPT-4 access be granted. 🚨

**PLEASE READ THIS**:

In order for a PR to be merged, it must fail on GPT-4. We are aware that right now, users do not have access, so you will not be able to tell if the eval fails or not. Please run your eval with GPT-3.5-Turbo, but keep in mind as we run the eval, if GPT-4 gets higher than 90% on the eval, we will likely reject it since GPT-4 is already capable of completing the task.

We plan to roll out a way for users submitting evals to see the eval performance on GPT-4 soon. Stay tuned! Until then, you will not be able to see the eval performance on GPT-4. **Starting April 10, the minimum eval count is 15 samples, we hope this makes it easier to create and contribute evals.**

Also, please note that we're using **Git LFS** for storing the JSON files, so please make sure that you move the JSON file to Git LFS before submitting a PR. Details on how to use Git LFS are available [here](https://git-lfs.com).

## Eval details 📑

### Eval name

state_capital_trivia

### Eval description

Some trivia questions about US states and state capitals.   With some word puzzle twists.

### What makes this a useful eval?

US states and capitals are well known facts, for example available from Wikipedia.   However with some very
simple word puzzles both GPT-3.5 and GPT-4 did not work well.   To answer these questions, GPT need to use 
the facts and then run some string matching and counting to solve the puzzle.

## Criteria for a good eval ✅

Below are some of the criteria we look for in a good eval. In general, we are seeking cases where the model does not do a good job despite being capable of generating a good response (note that there are some things large language models cannot do, so those would not make good evals).

Your eval should be:

- [x] Thematically consistent: The eval should be thematically consistent. We'd like to see a number of prompts all demonstrating some particular failure mode. For example, we can create an eval on cases where the model fails to reason about the physical world.
- [x] Contains failures where a human can do the task, but either GPT-4 or GPT-3.5-Turbo could not.
- [x] Includes good signal around what is the right behavior. This means either a correct answer for `Basic` evals or the `Fact` Model-graded eval, or an exhaustive rubric for evaluating answers for the `Criteria` Model-graded eval.
- [x] **Include at least 15 high-quality examples.**

If there is anything else that makes your eval worth including, please document it below.

### Unique eval value

> Insert what makes your eval high quality that was not mentioned above. (Not required)

## Eval structure 🏗️

Your eval should

- [x] Check that your data is in `evals/registry/data/{name}`
- [x] Check that your YAML is registered at `evals/registry/evals/{name}.yaml`
- [x] Ensure you have the right to use the data you submit via this eval

(For now, we will only be approving evals that use one of the existing eval classes. You may still write custom eval classes for your own cases, and we may consider merging them in the future.)

## Final checklist 👀

### Submission agreement

By contributing to Evals, you are agreeing to make your evaluation logic and data under the same MIT license as this repository. You must have adequate rights to upload any data used in an Eval. OpenAI reserves the right to use this data in future service improvements to our product. Contributions to OpenAI Evals will be subject to our usual Usage Policies (<https://platform.openai.com/docs/usage-policies>).

- [x] I agree that my submission will be made available under an MIT license and complies with OpenAI's usage policies.

### Email address validation

If your submission is accepted, we will be granting GPT-4 access to a limited number of contributors. Access will be given to the email address associated with the commits on the merged pull request.

- [x] I acknowledge that GPT-4 access will only be granted, if applicable, to the email address used for my merged pull request.

### Limited availability acknowledgment

We know that you might be excited to contribute to OpenAI's mission, help improve our models, and gain access to GPT-4. However, due to the requirements mentioned above and the high volume of submissions, we will not be able to accept all submissions and thus not grant everyone who opens a PR GPT-4 access. We know this is disappointing, but we hope to set the right expectation before you open this PR.

- [x] I understand that opening a PR, even if it meets the requirements above, does not guarantee the PR will be merged nor GPT-4 access be granted.

### Submit eval

- [ ] I have filled out all required fields of this form
- [ ] I have used **Git LFS** for the Eval JSON data
- [ ] (Ignore if not submitting code) I have run `pip install pre-commit; pre-commit install` and have verified that `black`, `isort`, and `autoflake` are running when I commit and push

Failure to fill out all required fields will result in the PR being closed.

### Eval JSON data

Since we are using Git LFS, we are asking eval submitters to add in as many Eval Samples (at least 5) from their contribution here:

<details>
  <summary>View evals in JSON</summary>

  ### Eval
  ```jsonl
  INSERT_EVAL_HERE
  ```
</details>

RishTheFish21 commented:
Not a reviewer, but you have to give the model more specific instructions on how the output is supposed to look like.

usama-openai reviewed (changes requested):
Thanks for the contribution. I would like to request some changes.

1. The evaluation method being used to evaluate the answer is `Match` which only works when an exact match is found. The user needs to provide very specific instructions to the model about the output format, and the ideal answer should be in that expected format. There are no specific instructions provided about the output format, and the model may provide an answer like `The state's name that ends with "Island" is Rhode Island`, but the ideal answer expects an output like `Rhode Island`. Due to this, the `Match` evaluation will mark the completion wrong despite the right answer being provided by the model.

    Instead of using `Match`, you need to use the `Includes` evaluation method.

2. There is no context provided for the model in the prompt. There are many countries in the world that have states. You need to specify that you are asking about US states.

3. The query `Which state X whose capital is X City` seems a little bit ambiguous. Can you make it more clear?

4. The following two samples have the same question but a different ideal answer.

    - `{"input": [{"role": "system", "content": "You are a helpful assistant."}, {"role": "user", "content": "Which state's capital is first in alphabit order"}], "ideal": "New York"}`
    - `{"input": [{"role": "system", "content": "You are a helpful assistant."}, {"role": "user", "content": "Which state's capital is first in alphabit order"}], "ideal": "New Jersey"}`

    Kindly add only unique samples with correct answers.

We would love to review the PR again after the suggested changes.

fengttt commented:
> Not a reviewer, but you have to give the model more specific instructions on how the output is supposed to look like.

Thank you.   Added system prompt and changed test to includes.

fengttt commented:
> Thanks for the contribution. I would like to request some changes.
> 
> 1. The evaluation method being used to evaluate the answer is `Match` which only works when an exact match is found. The user needs to provide very specific instructions to the model about the output format, and the ideal answer should be in that expected format. There are no specific instructions provided about the output format, and the model may provide an answer like `The state's name that ends with "Island" is Rhode Island`, but the ideal answer expects an output like `Rhode Island`. Due to this, the `Match` evaluation will mark the completion wrong despite the right answer being provided by the model.
>    Instead of using `Match`, you need to use the `Includes` evaluation method.
> 2. There is no context provided for the model in the prompt. There are many countries in the world that have states. You need to specify that you are asking about US states.
> 3. The query `Which state X whose capital is X City` seems a little bit ambiguous. Can you make it more clear?
> 4. The following two samples have the same question but a different ideal answer.
>    
>    * `{"input": [{"role": "system", "content": "You are a helpful assistant."}, {"role": "user", "content": "Which state's capital is first in alphabit order"}], "ideal": "New York"}`
>    * `{"input": [{"role": "system", "content": "You are a helpful assistant."}, {"role": "user", "content": "Which state's capital is first in alphabit order"}], "ideal": "New Jersey"}`
>    
>    Kindly add only unique samples with correct answers.
> 
> We would love to review the PR again after the suggested changes.



> Thanks for the contribution. I would like to request some changes.
> 
> 1. The evaluation method being used to evaluate the answer is `Match` which only works when an exact match is found. The user needs to provide very specific instructions to the model about the output format, and the ideal answer should be in that expected format. There are no specific instructions provided about the output format, and the model may provide an answer like `The state's name that ends with "Island" is Rhode Island`, but the ideal answer expects an output like `Rhode Island`. Due to this, the `Match` evaluation will mark the completion wrong despite the right answer being provided by the model.
>    Instead of using `Match`, you need to use the `Includes` evaluation method.
> 2. There is no context provided for the model in the prompt. There are many countries in the world that have states. You need to specify that you are asking about US states.
> 3. The query `Which state X whose capital is X City` seems a little bit ambiguous. Can you make it more clear?
> 4. The following two samples have the same question but a different ideal answer.
>    
>    * `{"input": [{"role": "system", "content": "You are a helpful assistant."}, {"role": "user", "content": "Which state's capital is first in alphabit order"}], "ideal": "New York"}`
>    * `{"input": [{"role": "system", "content": "You are a helpful assistant."}, {"role": "user", "content": "Which state's capital is first in alphabit order"}], "ideal": "New Jersey"}`
>    
>    Kindly add only unique samples with correct answers.
> 
> We would love to review the PR again after the suggested changes.

Thank you for the detailed feedback.   Fixed as suggested.

usama-openai reviewed (changes requested):
Thanks for updating the PR. There are still some issues with the samples.

1. For the question `How many states in US begin with word New`, the ideal answer is `4`, but the model may provide something like `Four states in the US begin with the word "New": New Hampshire, New Jersey, New Mexico, and New York.`. The evaluation method will mark such completion as wrong. You need to add proper instructions about the output format.

2. It seems like you didn't address the following change request properly:

    > The query `Which state X whose capital is X City` seems a little bit ambiguous. Can you make it more clear?

    You can modify this query as follows:

    `Write the name of the US state or states whose capital city also has the same name. For example, if the state name is 'X", then the capital city is "X city".`

3. The evaluation method being used is `Includes`, which checks if the ideal answer is present anywhere in the completion, and if proper output formatting is not used, it causes false positives. For example, for the query `How may state's capital is the lagest city by population in its state`, the ideal answer is `15`, and the model completion is `As of 2021, there are 17 state capitals that are also the largest city by population in their respective states. These capitals are: 1. Atlanta, Georgia 2. Boise, Idaho 3. Boston, Massachusetts 4. Charleston, West Virginia 5. Denver, Colorado 6. Des Moines, Iowa 7. Honolulu, Hawaii 8. Indianapolis, Indiana 9. Jackson, Mississippi 10. Little Rock, Arkansas 11. Nashville, Tennessee 12. Oklahoma City, Oklahoma 13. Phoenix, Arizona 14. Providence, Rhode Island 15. Salt Lake City, Utah 16. St. Paul, Minnesota 17. Wichita, Kansas. Please note, this information may change over time as populations fluctuate in cities and their metropolitan areas.`. As `15` is present in the completion, the evaluation method will mark this completion correct.

    For such queries, format the prompt like `How many state capitals are the largest cities by population in their state? After explanation, format the final numeric value as [number].`, and add the ideal answer as `"ideal": "[15]"` to eliminate false positives.

    Also, it seems like the ideal answer is not correct, and the correct answer is `17`. Kindly validate the ideal answers for all the samples.

4. There are too many typos and grammatical errors in the dataset. Kindly correct these errors.

5. The questions, like asking about population, don't age well with time because population changes with time, and the ideal answer may not remain correct after some years. For such queries, you need to add a timeframe to the query to eliminate such risks. For example, you can ask like this: `According to data available until August 2021, How many state capitals are the largest cities by population in their state? After explanation, format the final numeric value as [number].`

usama-openai commented:
Closing the PR due to inactivity; please feel free to reopen if you get a chance to address comments.
